File: launcher/android-launcher/android-launcher/src/main/java/org/asf/connective/tasks/AsyncTask.java
package org.asf.connective.tasks;

/**
 * 
 * Internal async task object
 * 
 * @author Sky Swimmer
 *
 */
public class AsyncTask {
	private Runnable action;
	private boolean run;

	public AsyncTask(Runnable action) {
		this.action = action;
	}

	void run() {
		try {
			action.run();
		} finally {
			run = true;
		}
	}

	public boolean hasRun() {
		return run;
	}

	public void block() {
		if (run)
			return;
		while (!run)
			try {
				Thread.sleep(1);
			} catch (InterruptedException e) {
			}
	}

}

File: feraltweaks-bootstrap/Modloader/Actions/Pooling/AsyncTask.cs
using System;
using System.Threading;

namespace FeralTweaks.Actions.Internal.AsyncTasks
{
    /// <summary>
    /// Async Task Container
    /// </summary>
    public class AsyncTask
    {
        private Action _action;
        private bool _run;
        
        /// <summary>
        /// Creates a task container
        /// </summary>
        /// <param name="action">Action assigned to the container</param>
        public AsyncTask(Action action)
        {
            _action = action;
        }

        internal void Run()
        {
            try
            {
                _action();
            }
            finally
            {
                _run = true;
            }
        }

        /// <summary>
        /// Checks if the task has been run already
        /// </summary>
        public bool HasRun
        {
            get
            {
                return _run;
            }
        }

        /// <summary>
        /// Waits for the task to finish running
        /// </summary>
        public void Block()
        {
            while (!_run)
                Thread.Sleep(1);
        }

    }
}